admin/plugins.php

```diff
@@ -390,3 +458,87 @@ function perflab_render_plugin_card( array $plugin_data ) {
 	</div>
 	<?php
 }
+
+// WordPress AJAX action to handle the button click event.
+add_action( 'wp_ajax_perflab_install_activate_standalone_plugins', 'perflab_install_activate_standalone_plugins_callback' );
+add_action( 'wp_ajax_nopriv_perflab_install_activate_standalone_plugins', 'perflab_install_activate_standalone_plugins_callback' );
+
+/**
+ * Handles the standalone plugin install and activation via AJAX.
+ *
+ * @since n.e.x.t
+ */
+function perflab_install_activate_standalone_plugins_callback() {
+	if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'perflab-install-activate-plugins' ) ) {
+		$status['errorMessage'] = esc_html__( 'Invalid nonce: Please refresh and try again.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
+	if ( ! isset( $_REQUEST['data'] ) ) {
+		$status['errorMessage'] = esc_html__( 'Invalid plugin data: Please refresh and try again.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
+	if ( ! function_exists( 'plugins_api' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	}
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	$plugins_to_install = $_REQUEST['data'];
+	$plugins_to_install = explode( ',', $plugins_to_install );
+
+	$get_standalone_plugins_module_map = perflab_get_standalone_plugins_module_map();
+	$get_standalone_plugins_file_map   = perflab_get_standalone_plugins_file_map();
+	$modules                           = perflab_get_module_settings();
+
+	$status = array();
+	foreach ( $plugins_to_install as $plugin_slug ) {
+
+		$api = perflab_query_plugin_info( $plugin_slug );
+
+		$module_slug = isset( $get_standalone_plugins_module_map[ $plugin_slug ] ) ? $get_standalone_plugins_module_map[ $plugin_slug ] : '';
+		$plugin_slug = $get_standalone_plugins_file_map[ $plugin_slug ];
```

Comment:
See my other comments, I think this can be simplified by relying on the existing function `perflab_get_standalone_plugins_constants()` and assuming that the plugin slug is the same as the module slug minus the focus area directory.

Reply:
Thanks @felixarntz for the feedback.

We use `perflab_get_standalone_plugins_file_map()` to get plugin main file to check if that plugin is installed (not activated) in WP or not.